docs/adrs/009-multus-thick-plugin.md

```diff
@@ -0,0 +1,41 @@
+# Multus thick plugin support in rke2
+
+Proposal date: 2025-06-18
+
+## Status
+
+## Context
+
+Multus is a k8s CNI multiplexer that allows the user to add several network interfaces to a pod.
+In 2023, the upstream Multus project introduced a new way to deploy Multus called "thick" plugin as opposed to the method already supported in rke2 called "thin" plugin.
+
+We have received [a request](https://github.com/harvester/harvester/issues/7042) from the Harvester team to support Multus thick plugin in rke2. 
+See also SURE-8203.
+
+In addition, the thick plugin deployment should become the default method from upstream in the future.
+
+### Pros
+- allows the use of [multus-dynamic-networks-controller](https://github.com/k8snetworkplumbingwg/multus-dynamic-networks-controller) by Harvester to implement [interface hotplugging](https://kubevirt.io/user-guide/network/hotplug_interfaces/) with Kubevirt
+- Multus thick plugin provides better metrics
+
+### Cons
+- higher resource consumption than with thin plugin
+- increased complexity to support the multus CRI shim
```

Comment:
Do you have an overview on what needs to happen to integrate it with containerd? If the containerd configuration needs to change, I think we need to restart containerd to consume the updated config. IIRC, that will restart rke2 process

Reply:
It does not look to me like the containerd config needs to change. The thick plugin implements the CRI itself, and then passes through to containerd. So we will need to start up both containerd and the thick plugin, point kubelet/crictl at that as the container runtime endpoint, and then the thick plugin talks to containerd's CRI endpoint.

Reply:
The thick plugin does not do anything with CRI no, but the dynamic network controller thank kube-virt needs does.

https://kubevirt.io/user-guide/network/hotplug_interfaces/#requirements
>  https://github.com/k8snetworkplumbingwg/multus-dynamic-networks-controller
> Multus Dynamic Networks Controller: this daemon will listen to annotation changes, and trigger Multus to configure a new attachment for this pod.

Reply:
I just did a test of the dynamic networks controller + thick plugin setup.
The dnc pod needs to have the CRI socket mounted but there is no other setup to do.

Then adding and deleting interfaces from a pod following the README works by adding annotations to the pod manifest which are watched by the dynamic network controller.

Reply:
Ok so it sounds like both can run in pods deployed by the chart then?